component/loki/source/gcplog/gcplog.go

```diff
@@ -0,0 +1,161 @@
+package gcplog
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/common/loki"
+	flow_relabel "github.com/grafana/agent/component/common/relabel"
+	gt "github.com/grafana/agent/component/loki/source/gcplog/internal/gcplogtarget"
+	"github.com/prometheus/prometheus/model/relabel"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name: "loki.source.gcplog",
+		Args: Arguments{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+// Arguments holds values which are used to configure the loki.source.gcplog
+// component.
+type Arguments struct {
+	// TODO(@tpaschalis) Having these types defined in an internal package
+	// means that an external caller cannot build this component's Arguments
+	// by hand for now.
+	PullTarget   *gt.PullConfig      `river:"pull,block,optional"`
+	PushTarget   *gt.PushConfig      `river:"push,block,optional"`
+	ForwardTo    []loki.LogsReceiver `river:"forward_to,attr"`
+	RelabelRules *flow_relabel.Rules `river:"relabel_rules,attr,optional"`
```

Comment:
Did we want to change this to struct based instead of pointer based?

Reply:
Yes, you're right. Change is pending on PR #2809.

Once that other PR is merged I'll rebase and amend this.